msmbuilder/preprocessing/normalize.py

```diff
@@ -0,0 +1,157 @@
+
+from __future__ import print_function, division, absolute_import
+import numpy as np
+import warnings
+from ..base import BaseEstimator
+from ..utils import check_iter_of_sequences, array2d
+from sklearn.base import TransformerMixin
+
+#-----------------------------------------------------------------------------
+# Code
+#-----------------------------------------------------------------------------
+
+class Normalize(BaseEstimator, TransformerMixin):
+    """Transform data to zero mean and unit variance
+
+
+    Attributes
+    ----------
+
+    means_ : array-like, shape (1, n_features)
+        Overall mean of each feature over all data.
+    stdevs_ : array-like, shape (1, n_features)
+        Overall standard deviation of each feature.
+    n_observations_ : int
+        Total number of data points fit by the model. Note that the model
+        is "reset" by calling `fit()` with new sequences, whereas
+        `partial_fit()` updates the fit with new data, and is suitable for
+        online learning.
+    n_sequences_ : int
+        Total number of sequences fit by the model. Note that the model
+        is "reset" by calling `fit()` with new sequences, whereas
+        `partial_fit()` updates the fit with new data, and is suitable for
+         online learning.
+
+    Note: Numerically stable method of computing variance adapted from:
+    https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
+    """
+
+    def __init__(self):
+
+        """
+        Initialize variables. Variance calculation is simpler with two passes over the data.
+            We require passing `sequences` to __init__() such that the total number of data
+            points `n` can be calculated here. This is needed for numerically stable 
+            calculation 
+
+        """
+        self.M = 0.
+        self.sum1 = 0.
+        self.sum2 = 0.
+        self.sum3 = 0.
+        
+    @property
+    def mean(self):
+        return self.M
+    @property
+    def std(self):
+        if self.n==1:
+            return float('nan')
+        return np.sqrt(self.var)
+
+    @property
+    def var(self):
+        if self.n==1:
+            return float('nan')
+        return (self.sum2 - np.square(self.sum3) / self.n) / (self.n-1.)
+
+    def fit(self, sequences):
+        """Fit the model with a collection of sequences.
+
+        This method is not online.  Any state accumulated from previous calls to
+        fit() or partial_fit() will be cleared. For online learning, use
+        `partial_fit`.
+
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        
+        check_iter_of_sequences(sequences, max_iter=3)  # we might be lazy-loading
+
+        self.k = len(sequences)
+        self.n = 0.
+        self.sum1 = 0.
+        self.sum2 = 0.
+        self.sum3 = 0.
+
+        #First Pass
+        for sequence in sequences:
+            self.n += float(np.shape(sequence)[0])
```

Comment:
python2 will truncate it to int (dumb python 2!). python3 will be nice. You can get the python3 behavior by putting

```python
from __future__ import division
```

at the top